posts/2021-01-22-jwtbuilder-redirects-21001.adoc

```diff
@@ -0,0 +1,182 @@
+
+New in Open Liberty 21.0.0.1, the `jwtBuilder` element has been enhanced with a new attribute called `nbfOffset` which can be used to configure an NBF claim for a JWT payload. The time set for the `nbfOffset` will be added to the current time and the result will determine when Json Web Tokens will start to be accepted for processing.
+
+To configure the "not-before" claim using `jwtBuilder`, add the following to your `sever.xml` configuration file.
+
+`<jwtBuilder nbfOffset="1800s" />`
+
+If the JWT was issued at the current time, then the token can only be used after 1800 seconds have passed from the current time.
+
+`jwtBuilder` is a part of the `jwt-1.0` feature, to add the feature to your project add the following to the `server.xml`:
+
+[source, xml]
+----
+<server>
+  <featureManager>
+    <feature>jwt-1.0</feature>
+  </featureManager>
+</server>
+----
+
+For more information:
+
+* link:https://openliberty.io/docs/21.0.0.1/reference/config/jwtBuilder.html[Open Liberty JWT Builder Documentation]
+
+* link:https://www.ibm.com/support/knowledgecenter/SSAW57_liberty/com.ibm.websphere.wlp.nd.multiplatform.doc/ae/twlp_sec_build_jwt.html[Building JSON Web Tokens in Liberty]
+
+[#redirects]
+== Relative linking for HTTP Redirects
+
+Included in Open Liberty 21.0.0.1 is an enhancement for the `servlet-4.0` feature, within Open Liberty the `sendRedirect()` is used to direct a client to a new page or location away from the original page. Previously, Open Liberty would always convert the provided relative URL in the `sendRedirect()` function to an absolute URL. This could lead to problems for applications that took advantage of reverse proxy servers.
+
+To solve the problems presented, Open Liberty 21.0.0.1 ahs introduced a new `redirecttorelativeurl` web container property that will tell the application whether or not to construct absoulte URLs from relative redirect links. This property can be set in the `server.xml` file:
+
+[source, xml]
+----
+<server>
+  ...
+    <webContainer redirecttorelativeurl="true"/>
+  ...
+</server>
+----
+
+For more information:
+
+* link:https://openliberty.io/docs/21.0.0.1/reference/feature/servlet-4.0.html[Open Liberty Servlet Documentation]
+
+[#conf]
+=== MicroProfile Config 2.0 enters general availability
+
+link:https://github.com/eclipse/microprofile-config[MicroProfile Config] enables application configuration from different locations (ConfigSources). MicroProfile Config 2.0 in Open Liberty consumes link:https://github.com/smallrye/smallrye-config[SmallRye's implementation] of the specification.
+
+MicroProfile Config 2.0 is a major version release (from 1.4) which introduces Functional and API/SPI changes in Microprofile Config 2.0.
+
+For a more in-depth look at the new capabilities offered by MicroProfile 2.0, take a look at the full explanation in the 20.0.0.12-beta blog post, link:https://openliberty.io/blog/2020/10/27/microprofile40-jakartaee9-beta-200012.html#conf[here].
```

Comment:
I don't think its MP 2.0 change to Microprofile config 2.0

Reply:
Thanks for spotting, changing.

Reply:
MP Config section has been removed, will reopen if the section is added back in.